puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -190,17 +190,34 @@ def build_events_list(events)
   # @api private
   def resource_status_to_hash(resource_status)
     defaulted_corrective_change = defined?(resource_status.corrective_change) ? resource_status.corrective_change : nil
-    {
-      "skipped"           => resource_status.skipped,
-      "timestamp"         => Puppet::Util::Puppetdb.to_wire_time(resource_status.time),
-      "resource_type"     => resource_status.resource_type,
-      "resource_title"    => resource_status.title.to_s,
-      "file"              => resource_status.file,
-      "line"              => resource_status.line,
-      "containment_path"  => resource_status.containment_path,
-      "corrective_change" => defaulted_corrective_change,
-      "events"            => build_events_list(resource_status.events),
-    }
+    if resource_event_index_size_reached?(resource_status)
+      Puppet.warning("Skipping resource_event on #{self.host} caused by Resource type: #{resource_status.resource_type} in #{resource_status.file}:#{resource_status.line}")
+    else
+      {
+        "skipped"           => resource_status.skipped,
+        "timestamp"         => Puppet::Util::Puppetdb.to_wire_time(resource_status.time),
+        "resource_type"     => resource_status.resource_type,
+        "resource_title"    => resource_status.title.to_s,
+        "file"              => resource_status.file,
+        "line"              => resource_status.line,
+        "containment_path"  => resource_status.containment_path,
+        "corrective_change" => defaulted_corrective_change,
+        "events"            => build_events_list(resource_status.events),
+      }
+    end
+  end
```

Comment:
The `if` should only include the error message, the hash with the resource should be sent as before.

Reply:
Updated. Also added a second condition to notify before the limit is reached.